risc0/cargo-risczero/src/commands/build_guest.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    fs::{self, File},
+    io::Write,
+    path::PathBuf,
+    process::Command,
+};
+
+use anyhow::{Context, Result};
+use cargo_metadata::MetadataCommand;
+use clap::Parser;
+use docker_generate::DockerFile;
+use risc0_zkvm::{sha::Digest, MemoryImage, Program, MEM_SIZE, PAGE_SIZE};
+use risc0_zkvm_platform::memory;
+
+use crate::utils::{ensure_binary, CommandExt};
+
+const DOCKER_IGNORE: &'static str = "**/elfs\n**/target\n**/Dockerfile\n**/.git";
```

Comment:
Sometimes in situations like these I use the `r#""#` multi-line string syntax with rust. That way you don't have to manually put in the \n characters. [ref](https://doc.rust-lang.org/reference/tokens.html#raw-string-literals)